database/scrapers/faq_script.ts

```diff
@@ -0,0 +1,190 @@
+import { DateTime } from "luxon";
+import { Readable } from "node:stream";
+
+import logger from "@adonisjs/core/services/logger";
+
+import GuideArticle from "#models/guide_article";
+import GuideQuestion from "#models/guide_question";
+import FilesService from "#services/files_service";
+
+interface GuideArticlesOld {
+  data: {
+    id: number;
+    name: string;
+    cover: string;
+    short_description: string;
+    description: string | null;
+    order: number;
+    questions: number[];
+  }[];
+}
+
+interface GuideQuestionsOld {
+  data: {
+    id: number;
+    status: string;
+    date_created: string;
+    date_updated: string;
+    question: string;
+    answer: string;
+    type: number;
+  }[];
+}
+
+interface PivotTable {
+  data: {
+    id: number;
+    FAQ_Types_id: number | null;
+    FAQ_id: number;
+    sort: number | null;
+  }[];
+}
+
+interface ImageMetadata {
+  data: {
+    type: string;
+  };
+}
+
+async function uploadImage(imageUrl: string) {
+  const filesService = new FilesService();
+
+  const [fetchedImage, fetchedImageMetadata] = await Promise.all([
+    fetch(`https://admin.topwr.solvro.pl/assets/${imageUrl}`),
+    fetch(`https://admin.topwr.solvro.pl/files/${imageUrl}`),
+  ]);
+
+  if (!fetchedImage.ok) {
+    throw new Error(
+      `Failed to fetch the image. HTTP status: ${fetchedImage.status}`,
+    );
+  }
+
+  if (!fetchedImageMetadata.ok) {
+    throw new Error(
+      `Failed to fetch the image metadata. HTTP status: ${fetchedImageMetadata.status}`,
+    );
+  }
+
+  const imageMetadata = (await fetchedImageMetadata.json()) as ImageMetadata;
+  const extension = imageMetadata.data.type.split("/")[1];
```

Comment:
this did not solve my previous concerns: you're basically still doing the same thing, you're still using the last part of the mime type which is not guaranteed to be a valid file extension.
extract the extension from one of the filenames that appear in the metadata instead